ai/core/src/unitycatalog/ai/core/oss.py

```diff
@@ -0,0 +1,647 @@
+        """
+        Delete a function by its full name.
+
+        This is a synchronous version of `delete_function_async`.
+
+        Args:
+            function_name: The full name of the function to delete.
+                It should be in the format of "catalog.schema.function_name".
+            timeout: The timeout in seconds.
+
+        Returns:
+            None
+        """
+
+        pass
+
+    @override
+    def to_dict(self) -> Dict[str, Any]:
+        return {k: getattr(self, k) for k in ["uc"] if getattr(self, k) is not None}
+
+
+class FunctionDefinition(NamedTuple):
+    """
+    A named tuple representing a dynamically constructed Python function.
+
+    Attributes:
+        function_def: The function definition as a string.
+        function_head: The function signature as a string.
+    """
+
+    function_def: str
+    function_head: str
+
+
+def dynamically_construct_python_function(function_info: FunctionInfo) -> FunctionDefinition:
+    """
+    Construct a Python function from the given FunctionInfo.
+
+    Args:
+        function_info: The FunctionInfo object containing the function metadata.
+
+    Returns:
+        A FunctionDefinition named tuple containing the function definition and function head.
+    """
+
+    param_names = []
+    if function_info.input_params and function_info.input_params.parameters:
+        param_names = [param.name for param in function_info.input_params.parameters]
+    function_head = f"{function_info.name}({', '.join(param_names)})"
+    func_def = f"def {function_head}:\n"
+    if function_info.routine_body == "EXTERNAL":
+        for line in function_info.routine_definition.split("\n"):
+            func_def += f"    {line}\n"
+    else:
+        raise NotImplementedError(f"routine_body {function_info.routine_body} not supported")
+
+    return FunctionDefinition(function_def=func_def, function_head=function_head)
+
+
+def validate_input_parameter(
+    parameter: Union[FunctionParameterInfo, Dict[str, str]],
+) -> FunctionParameterInfo:
+    """
+    Validate the input parameter and convert it to FunctionParameterInfo.
+
+    Args:
+        parameter: The input parameter to validate. If it is a dict, it will be converted to FunctionParameterInfo.
+
+    Returns:
+        The validated FunctionParameterInfo.
+    """
+
+    if isinstance(parameter, dict):
+        parameter = FunctionParameterInfo(**parameter)
+    elif not isinstance(parameter, FunctionParameterInfo):
+        raise TypeError(
+            f"Input parameter should be either a dict or an instance of "
+            f"FunctionParameterInfo, but got {type(parameter)}"
+        )
+
+    required_fields = ["name", "type_name", "type_text", "type_json"]
+
+    if missing_fields := [
+        field for field in required_fields if getattr(parameter, field, None) is None
+    ]:
+        raise ValueError(
+            f"Missing required fields in input parameter '{getattr(parameter, 'name', '')}': {missing_fields}."
```

Comment:
```suggestion
            f"Missing required fields in input parameter '{parameter}': {missing_fields}."
```
In case the parameter doesn't contain name?

Reply:
great catch!